app/src/main/java/com/vultisig/wallet/ui/navigation/NavGraph.kt

```diff
@@ -91,11 +91,23 @@ internal fun SetupNavGraph(
             Setup(navController)
         }
 
-        composable(route = Screen.KeygenFlow.route) {
+        composable(
+            route = Screen.KeygenFlow.route,
+            arguments = listOf(navArgument(Screen.KeygenFlow.ARG_VAULT_NAME) {
+                type = NavType.StringType
+                defaultValue = Screen.KeygenFlow.DEFAULT_NEW_VAULT
+            })
+        ) { navBackStackEntry ->
             val vaultDB = VaultDB(context)
             val allVaults = vaultDB.selectAll()
-            // TODO: later on will need to deal with reshare
-            KeygenFlowView(navController, Vault("New Vault ${allVaults.size + 1}"))
+            val vaultId = navBackStackEntry.arguments?.getString("vault_name") ?: ""
```

Comment:
Probably can reuse constant for argument name

```suggestion
            val vaultId = navBackStackEntry.arguments?.getString(Screen.KeygenFlow.ARG_VAULT_NAME) ?: ""
```